File: source/sat/src/main/java/ch/unisg/sat/controller/http/dto/request/aggregate/ArchitecturalKnowledgeRequestDto.java
package ch.unisg.sat.controller.http.dto.request.aggregate;

import ch.unisg.sat.controller.http.dto.request.entity.concept.IssueRequestDto;
import lombok.Getter;

@Getter
public class ArchitecturalKnowledgeRequestDto {

    public static final String MEDIA_TYPE = "application/architectural-knowledge+json";

    private IssueRequestDto issue;
    private AlternativeAggregateRequestDto[] alternatives;
}

File: source/frontend/src/lib/dto/request/aggregate/ArchitecturalKnowledgeRequestDto.ts
import type {IssueRequestDto} from "$lib/dto/request/entity/IssueRequestDto";
import type {AlternativeAggregateRequestDto} from "$lib/dto/request/aggregate/AlternativeAggregateRequestDto";

export class ArchitecturalKnowledgeRequestDto {
    issue: IssueRequestDto;
    alternatives: AlternativeAggregateRequestDto[];

    constructor(issue : IssueRequestDto, alternatives : AlternativeAggregateRequestDto[]) {
        this.issue = issue;
        this.alternatives = alternatives;
    }

    static create(issue : IssueRequestDto, alternatives : AlternativeAggregateRequestDto[]) {
        return new ArchitecturalKnowledgeRequestDto(issue, alternatives);
    }
}